### PR TITLE
Add documentation about deep mode

### PR DESCRIPTION
**Pull request by eliecharra**
For https://github.com/cloudskiff/driftctl/issues/819

image

### CODE REVIEW

**Review by sundowndev — APPROVED** [review]

**Review by lotoussa — APPROVED** [review]

File: docs/usage/cmd/scan.mdx [hunk 1]
```diff
@@ -318,6 +319,33 @@ For now, resources include:
...
+Deep mode enable resource detail retrieval. It was the original driftctl behavior.
+
+In **deep** mode we compare resources details to expected ones (like a terraform plan).
+In **non-deep** (the default) we only enumerate resources and compute which ones are out of IaC scope.
+
+Since it overlap the new terraform plan behavior (from 0.15 terraform plan shows diff between your state, and the remote) we moved the original behavior under the `--deep` **experimental** flag.
+
+
+:::info
+In driftctl version < 0.13 the deep mode was the default behavior. If you want to keep the old behavior in the newer version you have to enable the deep mode flag.
```
Comment (wbeuil) [inline docs/usage/cmd/scan.mdx:339]: `If you use a version of driftctl prior to 0.13, the deep ...`
`in a newer version ...`

**Review by wbeuil — CHANGES_REQUESTED** [review]

File: docs/providers/github/resources.mdx [hunk 2]
```diff
@@ -3,8 +3,10 @@ id: resources
 title: Supported Resources
 ---
 
-- [x] github_repository
-- [x] github_team
-- [x] github_membership
```
[the remark this answers is not in the document]
Comment (eliecharra) [reply docs/providers/github/resources.mdx:8]: Nice catch

File: docs/usage/cmd/scan.mdx [hunk 3]
```diff
@@ -318,6 +319,32 @@ For now, resources include:
...
+⚠️ This is **EXPERIMENTAL**
+
+:::warning
+Enabling deep mode while using a Terraform state as IaC source can lead to unexpected behaviors: false positive drifts, undetected drifts.
+:::
+
+Deep mode enables resources details retrieval. It was the original driftctl behavior.
+
+In **deep** mode we compare resources details to expected ones (like a terraform plan).
+In **non-deep** (the default one) we only enumerate resources and display which ones are out of IaC scope.
```
Comment (wbeuil) [inline docs/usage/cmd/scan.mdx:333]: Can you add the word `mode` just after non-deep to emphasize what it actually is?

File: docs/usage/cmd/scan.mdx [hunk 4]
```diff
@@ -318,6 +319,32 @@ For now, resources include:
...
+
+Since it overlaps the new terraform plan behavior (as of Terraform 0.15 it shows diffs between your state and the remote) we moved the original behavior under the `--deep` **experimental** flag.
```
Comment (wbeuil) [inline docs/usage/cmd/scan.mdx:335]: Put the command `terraform plan` between backticks.

**Review by wbeuil — CHANGES_REQUESTED** [review]

**Review by wbeuil — APPROVED** [review]

**Review by wbeuil — APPROVED** [review]

**Pull request merged**